packages/logo-grid/partials/dialog-tooltip/index.tsx

```diff
@@ -0,0 +1,195 @@
+ * the tooltip trigger and tooltip popup,
+ * render the tooltip centered and below
+ * the trigger.
+ *
+ * Allow viewport collisions to override
+ * the centered position where needed,
+ * using the collisionBuffer argument
+ * to inset the collision area so the tooltip
+ * doesn't appear at the very edge of the
+ * viewport.
+ */
+function calcDialogPosition(
+  /** DOMRect of the element that triggered the dialog */
+  triggerRect: DOMRect,
+  /** DOMRect of the dialog element itself */
+  tooltipRect: DOMRect,
+  /** Minimum distance in pixels that the dialog should be from the viewport edge  */
+  collisionBuffer: number,
+  /** Size in pixels of the "arrow" triangle tip of the dialog.
+   * This needs to be accounted for to achieve accurate vertical positioning.  */
+  arrowSize: number
+) {
+  if (!triggerRect || !tooltipRect) return { left: 0, top: 0 }
+  const triggerCenter = triggerRect.left + triggerRect.width / 2
+  const left = triggerCenter - tooltipRect.width / 2
+  const maxLeft = window.innerWidth - tooltipRect.width - collisionBuffer
+  return {
+    left: Math.min(Math.max(collisionBuffer, left), maxLeft),
+    top: triggerRect.bottom + arrowSize,
+  }
+}
+
+/**
+ * Given the bounding rectangle for
+ * the tooltip trigger, render a small
+ * triangular arrow.
+ *
+ * This arrow is centered relative to
+ * the trigger, but accounts for possible
+ * viewport collisions, as we would prefer
+ * to have the arrow connected to the popup
+ * (which is bound by the viewport) rather
+ * than have it perfectly centered but
+ * disconnected from the popup.
+ */
+function DialogArrow({
+  shown,
+  triggerRect,
+  collisionBuffer,
+  arrowSize,
+  theme,
+}: {
+  /** Whether the arrow should be shown */
+  shown: boolean
+  /** DOMRect of the target element, for positioning the arrow */
+  triggerRect: DOMRect
+  /** Minimum distance in pixels that the arrow should be from the viewport edge  */
+  collisionBuffer: number
+  /** Size in pixels of the arrow */
+  arrowSize: number
+  /** Arrow coloration  */
+  theme: 'light' | 'dark'
+}): React.ReactElement {
+  if (!shown) return null
+  const arrowLeft = triggerRect
+    ? `${Math.min(
+        // Centered position, covers most use cases
+        triggerRect.left - arrowSize + triggerRect.width / 2,
+        // Ensure the arrow is not rendered even partially offscreen,
+        // as it will look disconnected from our tooltip body,
+        // which must be rendered within the viewport
+        window.innerWidth - arrowSize * 2 - collisionBuffer
+      )}px`
+    : 'auto'
+  const arrowTop = triggerRect ? `${triggerRect.bottom}px` : 'auto'
+
+  return (
+    <Portal>
+      <div
+        className={classNames(s.dialogArrow, s[theme])}
+        style={
+          {
+            '--left': arrowLeft,
+            '--top': arrowTop,
+            '--arrow-size': arrowSize + 'px',
+          } as React.CSSProperties
+        }
+      />
+    </Portal>
+  )
+}
```

Comment:
My only thoughts here are that this is a lot of complicated custom code - I'm not sure that its avoidable, it almost feels like we removed tippy only to effectively re-implement it manually